thread/thread.h

```diff
@@ -212,9 +212,9 @@ namespace photon
     protected:
         int wait(Timeout timeout = {});
         int wait_defer(Timeout Timeout, void(*defer)(void*), void* arg);
-        void resume(thread* th, int error_number = ECANCELED);  // `th` must be waiting in this waitq!
-        int resume_all(int error_number = ECANCELED);
-        thread* resume_one(int error_number = ECANCELED);
+        void resume(thread* th, int error_number = -1);  // `th` must be waiting in this waitq!
+        int resume_all(int error_number = -1);
+        thread* resume_one(int error_number = -1);
         waitq() = default;
```

Comment:
What is the default behavior of the default value -1. Documents should be adjusted in:

https://github.com/alibaba/PhotonLibOS/blob/main/doc/docs/api/thread.md#thread_interrupt

https://github.com/alibaba/PhotonLibOS/blob/main/doc/i18n/cn/docusaurus-plugin-content-docs/current/api/thread.md#thread_interrupt

Reply:
If it comes from the inside of the synchronization primitives due to technical reasons, we use -1 to distinguish from user generated ordinary error numbers (>0). And 0 has already been used for another scenario.

Reply:
> What is the default behavior of the default value -1 ? Documents should be adjusted in:
> 
> https://github.com/alibaba/PhotonLibOS/blob/main/doc/docs/api/thread.md#thread_interrupt
> 
> https://github.com/alibaba/PhotonLibOS/blob/main/doc/i18n/cn/docusaurus-plugin-content-docs/current/api/thread.md#thread_interrupt

Those documents are about `photon::thread_interrupt`. The default errno in `photon::thread_interrupt` is not changed, so it still EINTR, nothing has changed in document.

Here is all about method `waitq::resume`, which is protected method, not a part of API.